src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/relationships-property/new-relationships.component.ts/new-relationships.component.ts

```diff
@@ -0,0 +1,98 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { DotMessageService } from '@services/dot-messages-service';
+import { PaginatorService } from '@services/paginator';
+import { ContentType } from '@portlets/content-types/shared/content-type.model';
+import { RelationshipCardinality } from '@portlets/content-types/fields/shared/relationship-cardinality.model';
+import { RelationshipService } from '@portlets/content-types/fields/service/relationship.service';
+import { DotContentletService } from '@services/dot-contentlet/dot-contentlet.service';
+import { Observable } from 'rxjs';
+
+@Component({
+    providers: [PaginatorService],
+    selector: 'dot-new-relationships',
+    templateUrl: './new-relationships.component.html'
+})
+export class NewRelationshipsComponent implements OnInit {
+    @Input()
+    cardinalityIndex: number;
+
+    @Input()
+    velocityVar: string;
+
+    @Input()
+    editing: boolean;
+
+    @Output()
+    change: EventEmitter<any> = new EventEmitter();
+
+    contentTypeCurrentPage: Observable<ContentType[]>;
+    relationshpsCardinalities: RelationshipCardinality[];
+
+    cardinality: RelationshipCardinality;
+    contentType: ContentType;
+
+    i18nMessages: {
+        [key: string]: string;
+    } = {};
+
+    constructor(
+        public dotMessageService: DotMessageService,
+        public contentTypePaginatorService: PaginatorService,
+        private relationshipService: RelationshipService,
+        private contentTypeService: DotContentletService) {
+
+    }
+    ngOnInit() {
+        this.dotMessageService
+            .getMessages([
+                'contenttypes.field.properties.relationship.new.label',
+                'contenttypes.field.properties.relationship.new.content_type.placeholder'
+            ])
+            .subscribe((res) => {
+                this.i18nMessages = res;
+            });
+
+        this.contentTypePaginatorService.url = 'v1/contenttype';
+
+        this.relationshipService.loadCardinalities().subscribe((cardinalities: RelationshipCardinality[]) => {
+            this.relationshpsCardinalities = cardinalities;
+            this.cardinality = this.relationshpsCardinalities[this.cardinalityIndex];
+        });
+
+        if (this.velocityVar) {
+            this.contentTypeService.getContentType(this.velocityVar).subscribe((contentType) => {
+                this.contentType = contentType;
```

Comment:
Can't this be an Observable and use async pipe? If not, you need to unsubscribe somehow